client/components/planning-editor-standalone/profile.ts

```diff
@@ -52,108 +71,173 @@ interface IFieldDefinition {
+                };
+
+                const field: IAuthoringFieldV2 = {
+                    id: id,
+                    name: gettext('Attached files'),
+                    fieldType: 'files',
+                    fieldConfig: fieldConfig,
+                };
+
+                return field;
             },
-            retrieveStoredValue: (storageValue: Array<IVocabularyItem>) => storageValue.map(({qcode}) => qcode),
         },
-    });
-
-    result.push({
-        fieldId: 'coverages',
-        getField: ({id, required}) => {
-            const fieldConfig: ICommonFieldConfig = {
-                required,
-            };
-
-            const field: IAuthoringFieldV2 = {
-                id: id,
-                name: gettext('Coverages'),
-                fieldType: 'coverages',
-                fieldConfig: fieldConfig,
-            };
-
-            return field;
+        {
+            fieldId: 'place',
+            getField: ({id, required}) => {
+                const fieldConfig: IDropdownConfigVocabulary = {
+                    source: 'vocabulary',
+                    vocabularyId: 'locators',
+                    multiple: true,
+                    required: required,
+                };
+
+                const field: IAuthoringFieldV2 = {
+                    id: id,
+                    name: gettext('Place'),
+                    fieldType: 'dropdown',
+                    fieldConfig: fieldConfig,
+                };
+
+                return field;
+            },
+            storageAdapter: {
+                storeValue: (operationalValue: Array<string>): Array<IVocabularyItem> => {
+                    const vocabulary = superdeskApi.entities.vocabulary.getAll().get('locators');
+                    const vocabularyItems = new Map<IVocabularyItem['qcode'], IVocabularyItem>(
+                        vocabulary.items.map((item) => [item.qcode, item]),
+                    );
+
+                    return operationalValue.map((qcode) => vocabularyItems.get(qcode));
+                },
+                retrieveStoredValue: (storageValue: Array<IVocabularyItem>) => storageValue.map(({qcode}) => qcode),
+            },
         },
-    });
+        {
+            fieldId: 'coverages',
+            getField: ({id, required}) => {
+                const fieldConfig: ICommonFieldConfig = {
+                    required,
+                };
+
+                const field: IAuthoringFieldV2 = {
+                    id: id,
+                    name: gettext('Coverages'),
+                    fieldType: 'coverages',
+                    fieldConfig: fieldConfig,
+                };
+
+                return field;
+            },
+        }
+    ];
+    const planningProfile = planningApi.contentProfiles.get('planning');
+    const customVocabularyIds =
+        (planningProfile.schema?.['custom_vocabularies'] as IProfileSchemaTypeList)?.vocabularies;
+
+    if ((customVocabularyIds?.length ?? 0) > 0) {
```

Comment:
custom vocabularies definition is quite big, could you move it to a separate file? here you would import the function, call it and push the result to the array of fields